cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java

```diff
@@ -167,7 +167,17 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
 
     private void doHandleEmbedded(CommandContext ctx, ModelControllerClient client) throws CommandLineException {
 
+        assert(embeddedServerRef != null);
+        assert(embeddedServerRef.get() != null);
+
         final ModelNode op = this.buildRequestWithoutHeaders(ctx);
+        if (embeddedServerRef.get().isHostController()) {
```

Comment:
WDYT about adding some validation in this block?

```
final ParsedCommandLine args = ctx.getParsedCommandLine();
if (this.adminOnly.isPresent(args) && "FALSE".equalsIgnoreCase(this.adminOnly.getValue(args)) {
 throw someExceptionCauseWeCantDoWhatTheyRequested;
}
```

Bonus is overriding the canAppearNext() method of adminOnly so it returns false if embeddedServerRef.get() != null && embeddedServerRef.get().isHostController(). That will remove that param from tab completion.